lib/inspector.js

```diff
@@ -108,7 +109,7 @@ class Session extends EventEmitter {
    * @param {string} method
    * @param {Record<unknown, unknown>} [params]
    * @param {Function} [callback]
-   * @returns {void}
+   * @returns {void | Promise}
```

Comment:
I believe it might make sense to add overloading types, by using jsdoc `@type` with something inline of:

```jsdoc
/**
 * @type {{
 * (method:string, params:Record<unknown, unknown>, callback:Function) => void;
 * (method:string, params:Record<unknown) => Promise;
 * }}
 */
```

But I'm honestly not sure how this was done for other functions that support overloading. I'm just curious about the TypeScript counterpart for these types when on `types/node`.

**Note.:** My comment is just a nitpick, not an actual blocker. Cheers!

Reply:
There's also the other way of inferencing within jsdoc

```jsdoc
@returns {callback extends Function ? void : Promise}
```